src/components/lab/canvas/molecules/CanvasMethane.js

```diff
@@ -31,38 +37,62 @@ const CanvasMethane = ({ x, y, shouldOscillate, oscillationFormula }) => {
   const hydrogenAtomRadius = CANVAS_ATOM_DIMENSIONS[HYDROGEN.size];
   const carbonAtomRadius = CANVAS_ATOM_DIMENSIONS[CARBON.size];
   const carbonAtomCenterPoint = {
-    x: shouldOscillate ? x + CARBON_AMPLITUDE * oscillationFormula : x,
+    x: shouldOscillate
+      ? x + oscillationDirection * CARBON_AMPLITUDE * sinusoidalOscillationPoint
+      : x,
     y,
   };
   const topLeftHydrogenAtomCenterPoint = {
     x: shouldOscillate
       ? x -
         CANVAS_METHANE_TOP_LEFT_HYDROGEN_ATOM_X_ADJUSTMENT_FACTOR +
-        TOP_LEFT_HYDROGEN_AMPLITUDE * oscillationFormula
+        oscillationDirection *
+          TOP_LEFT_HYDROGEN_AMPLITUDE *
+          sinusoidalOscillationPoint
       : x - CANVAS_METHANE_TOP_LEFT_HYDROGEN_ATOM_X_ADJUSTMENT_FACTOR,
     y: y - carbonAtomRadius - hydrogenAtomRadius,
   };
   const topRightHydrogenAtomCenterPoint = {
     x: shouldOscillate
       ? x +
         CANVAS_METHANE_TOP_RIGHT_HYDROGEN_ATOM_X_ADJUSTMENT_FACTOR +
-        TOP_RIGHT_HYDROGEN_AMPLITUDE * oscillationFormula
+        oscillationDirection *
+          TOP_RIGHT_HYDROGEN_AMPLITUDE.X *
+          sinusoidalOscillationPoint
       : x + CANVAS_METHANE_TOP_RIGHT_HYDROGEN_ATOM_X_ADJUSTMENT_FACTOR,
-    y: y - carbonAtomRadius - 0.25 * hydrogenAtomRadius,
+    y: shouldOscillate
+      ? y -
+        carbonAtomRadius -
+        0.25 * hydrogenAtomRadius +
+        oscillationDirection *
+          TOP_RIGHT_HYDROGEN_AMPLITUDE.Y *
+          sinusoidalOscillationPoint
+      : y - carbonAtomRadius - 0.25 * hydrogenAtomRadius,
   };
   const bottomRightHydrogenAtomCenterPoint = {
     x: shouldOscillate
       ? x +
         CANVAS_METHANE_BOTTOM_RIGHT_HYDROGEN_ATOM_X_ADJUSTMENT_FACTOR +
-        BOTTOM_RIGHT_HYDROGEN_AMPLITUDE * oscillationFormula
+        oscillationDirection *
+          BOTTOM_RIGHT_HYDROGEN_AMPLITUDE.X *
+          sinusoidalOscillationPoint
       : x + CANVAS_METHANE_BOTTOM_RIGHT_HYDROGEN_ATOM_X_ADJUSTMENT_FACTOR,
-    y: y + carbonAtomRadius + 0.25 * hydrogenAtomRadius,
+    y: shouldOscillate
+      ? y +
+        carbonAtomRadius +
+        0.25 * hydrogenAtomRadius +
```

Comment:
These `0.25` should probably be factored out. Maybe consider factoring out the formulas too, if possible?

Reply:
:+1: agree that these are not very readable. made some adjustments that cut some code and hopefully make this as readable as possible.

Reply:
(if you can think of other approaches I'd be really interested to see them!)